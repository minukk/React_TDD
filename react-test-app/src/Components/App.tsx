import React from "react";
import styled, { keyframes } from "styled-components";
import logo from "./logo.svg";

function App() {
  return (
    <Container>
      <Header>
        <AppLogo src={logo} alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <AppLink
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </AppLink>
      </Header>
    </Container>
  );
}

const Container = styled.div`
  text-align: cetner;
`;

const Header = styled.header`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  min-height: 100vh;
  background-color: #282c34 !important;
  font-size: calc(10px + 2vmin);
  color: #fff;
`;

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const AppLogo = styled.img`
  height: 40vmin;
  pointer-events: none;

  @media (prefers-reduced-motion: no-preference) {
    animation: ${spin} infinite 20s linear;
  }
`;

const AppLink = styled.a`
  color: #61dafb;
`;

export default App;
