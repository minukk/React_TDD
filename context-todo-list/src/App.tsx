import React from 'react';
import styled from 'styled-components';
import { ToDoList, InputContainer } from 'Components';
import { ToDoListProvider } from 'Context';

function App() {
  return (
    <ToDoListProvider>
      <Container>
        <Contents>
          <ToDoList />
          <InputContainer />
        </Contents>
      </Container>
    </ToDoListProvider>
  );
}

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  min-height: 100vh;
`;

const Contents = styled.div`
  display: flex;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  background-color: #fff;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;

export default App;
