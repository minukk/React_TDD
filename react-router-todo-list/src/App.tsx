import React from 'react';
import styled from 'styled-components';

import { ToDoListProvider } from 'Context';
import { Route, Routes } from 'react-router-dom';
import { List, Add, Detail, NotFound } from 'Pages';
import { PageHeader } from 'Components';

function App() {
  return (
    <ToDoListProvider>
      <Container>
        <PageHeader />
        <Routes>
          <Route path="/" element={<List />} />
          <Route path="/add" element={<Add />} />
          <Route path="/detail/:id" element={<Detail />} />
          <Route path="*" element={<NotFound />}></Route>
        </Routes>
      </Container>
    </ToDoListProvider>
  );
}

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  min-height: 100vh;
  background-color: #eee;
`;

export default App;
