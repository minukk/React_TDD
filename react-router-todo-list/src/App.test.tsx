import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import App from './App';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

describe('<App />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/');

    const { container } = render(
      <Router navigator={history} location={history.location}>
        <App />
      </Router>,
    );

    const header = screen.getByText('할 일 목록');
    expect(header).toBeInTheDocument();

    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
    expect(toDoList.firstChild).toBeNull();

    const input = screen.getByText('+');
    expect(input).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  // it('adds and deletes ToDo items', () => {
  //   const history = createMemoryHistory();
  //   history.push('/');

  //   render(
  //     <Router navigator={history} location={history.location}>
  //       <App />
  //     </Router>,
  //   );

  //   const input = screen.getByPlaceholderText('할 일을 입력해 주세요.');
  //   const button = screen.getByText('추가');
  //   fireEvent.change(input, { target: { value: 'study react 1' } });
  //   fireEvent.click(button);

  //   const todoItem = screen.getByText('study react 1');
  //   expect(todoItem).toBeInTheDocument();

  //   const deleteButton = screen.getByText('삭제');
  //   expect(deleteButton).toBeInTheDocument();

  //   const toDoList = screen.getByTestId('toDoList');
  //   expect(toDoList.childElementCount).toBe(1);

  //   fireEvent.change(input, { target: { value: 'study react 2' } });
  //   fireEvent.click(button);

  //   const todoItemSecond = screen.getByText('study react 2');
  //   expect(todoItemSecond).toBeInTheDocument();
  //   expect(toDoList.childElementCount).toBe(2);

  //   const deleteButtons = screen.getAllByText('삭제');
  //   fireEvent.click(deleteButtons[0]);

  //   expect(todoItem).not.toBeInTheDocument();
  //   expect(toDoList.childElementCount).toBe(1);
  // });

  it('does not add empty ToDo', () => {
    const history = createMemoryHistory();

    render(
      <Router navigator={history} location={history.location}>
        <App />
      </Router>,
    );

    const toDoList = screen.getByTestId('toDoList');
    const length = toDoList.childElementCount;

    const button = screen.getByText('+');
    fireEvent.click(button);

    expect(toDoList.childElementCount).toBe(length);
  });

  it('loads localStorage data', () => {
    localStorage.setItem('ToDoList', '["ToDo 1", "ToDo 2", "ToDo 3"]');
    const history = createMemoryHistory();

    render(
      <Router navigator={history} location={history.location}>
        <App />
      </Router>,
    );

    expect(screen.getByText('ToDo 1')).toBeInTheDocument();
    expect(screen.getByText('ToDo 2')).toBeInTheDocument();
    expect(screen.getByText('ToDo 3')).toBeInTheDocument();
    expect(screen.getAllByText('삭제').length).toBe(3);
  });
});
