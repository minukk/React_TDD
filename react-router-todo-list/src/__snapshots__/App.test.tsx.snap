// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
.c4 {
  min-width: 350px;
  height: 400px;
  margin-bottom: 20px;
  border: 1px solid #bdbdbd;
  overflow-y: scroll;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1e40ff;
}

.c2 {
  padding: 20px;
  color: #fff;
  font-size: 20px;
  font-weight: 600;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  background-color: #fff;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

.c5 {
  position: absolute;
  bottom: -30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 60px;
  border-radius: 30px;
  background-color: #304ffe;
  font-size: 20px;
  color: #fff;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c5:hover {
  background-color: #1e40ff;
}

.c5:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  background-color: #eee;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        할 일 목록
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
        data-testid="toDoList"
      />
      <a
        class="c5"
        href="/add"
      >
        +
      </a>
    </div>
  </div>
</div>
`;
